stack/manager.go

```diff
@@ -130,11 +136,40 @@ func (m *Manager) ListChanged(gitBaseRef string) (*Report, error) {
 		return nil, errors.E(errListChanged, err)
 	}
 
-	changedFiles, err := m.changedFiles(gitBaseRef)
+	_, err = m.changedFiles(cfg.BaseRef)
```

Comment:
I think this is hard to reason about, the memoization is just an internal thing, but here the caller relies on that behaviour.

Would it be possible to add `checks.UncommittedFiles` and `checks.UntrackedFiles` as params to m.changedFiles (with the option of them being empty), then do the appending in there?